blockchain/blockdao/blockdao.go

```diff
@@ -58,23 +66,38 @@ type (
 	}
 )
 
-// NewBlockDAO instantiates a block DAO
-func NewBlockDAO(indexers []BlockIndexer, cfg db.Config, deser *block.Deserializer) BlockDAO {
-	blkStore, err := filedao.NewFileDAO(cfg, deser)
-	if err != nil {
-		log.L().Fatal(err.Error(), zap.Any("cfg", cfg))
+// NewBlockDAOWithIndexersAndCache returns a BlockDAO with indexers which will consume blocks appended, and
+// caches which will speed up reading
+func NewBlockDAOWithIndexersAndCache(blkStore BlockDAO, indexers []BlockIndexer, cacheSize int) BlockDAO {
```

Comment:
may just change func name to `NewBlockDAOWithIndexersAndCache`, keep input param list the same
so we don't need to call explicitly `fileDAO, err := filedao.NewFileDAO(cfg, deser)` upfront, then pass `fileDAO` and call this.